functions/src/modules/SQR.functions.ts

```diff
@@ -20,52 +20,58 @@ export enum ISoundQualityReportSheet {
   Submissions = 'Submissions',
 }
 
-/////////////////////////////////////////////////
-//          OnNewAllotment (DB create and update Trigger)
-//      1. Mark the files in the database --> { status: "Given" }
-//              Function --> updateFilesOnNewAllotment
-//
-//      2. Send an email to the assignee to notify them of the new allotments
-//              Function --> sendEmailOnNewAllotment
-/////////////////////////////////////////////////
-export const updateFilesOnNewAllotment = functions.database
-  .ref('/sqr/allotments/{allotment_id}')
-  .onCreate((snapshot, context) => {
-    const allotment = snapshot.val();
-    let newDocKey = snapshot.key;
-
-    // loop through the FILES array in the NEW ALLOTMENT object
-    // and update their corresponding file objects
-    allotment.files.forEach(async file => {
-      let sqrRef = db.ref(
-        `/files/${allotment.list}/${file}/soundQualityReporting`
-      );
 
-      let sqrError = await sqrRef.update({
-        status: 'Given',
-        assignee: allotment.assignee,
-        timestampGiven: Math.round(new Date().getTime() / 1000),
-        timestampDone: null,
-      });
+/**
+ * Process registration of webform data
+ * 1. Registers user by transferring data into /users/ path
+ * 2. Check if users already exists
+ * 
+ * @function processRegistration()
+ */
+export const processRegistration = functions.database
+.ref('/webforms/registration/{registration_id}')
+.onCreate(async (snapshot, context) => {
+  const user = snapshot.val();
 
-      if (sqrError == undefined) {
-        // if Successful FILE Update, update the ALLOTMENT accordingly
+  const olduser = await db.ref("/users/").orderByChild("/emailAddress")
+                  .equalTo(user.email_address).once("value");
 
-        // case 1 -- the allotmnet is read from the spreadsheet
-        if (Object.keys(allotment).indexOf('sendNotificationEmail') > -1)
-          db.ref(`/sqr/allotments/${newDocKey}`).update({
-            filesAlloted: true,
-          });
-        // case 2 -- the allotmnet is inputted manually
-        else
-          db.ref(`/sqr/allotments/${newDocKey}`).update({
-            filesAlloted: true,
-            sendNotificationEmail: true,
-          });
-      }
-    })
+  // If the user already exists stop the execution and log the error
+  if (olduser.val() !== undefined) {
+    console.log(`Error! User already exist with email: ${user.email_address}`)
+    return false;
   }
-)
+
+  // if user email doesn't exists prepare the new_user Object
+  const new_user: any = {
+    timestamp: moment.format('x'),
+    emailAddress: user.email_address,
+    phoneNumber: user.contact_number,
+    isAvailableOnWhatsApp: user.i_am_available_on_whatsapp,
+    location: user.country,
+    name: user.your_name,
+    services: (<any>Object).values(user.seva).join(", "),
```

Comment:
I cannot understand the logic of this transformation. I think that there is no transformation needed at all.

Reply:
Anyhow, this transformation is not needed here.

Reply:
@anantakrishna See this is the error we get if we remove that type transformation.

![image](https://user-images.githubusercontent.com/10194373/50822867-8c16dd00-1354-11e9-94ca-223a8fdd10fe.png)

Reply:
OK, I mistakenly was thinking that `webform.seva` is a string. No issues here. Thank you.